src/Source/DoctrineDBALConnectionSourceIterator.php

```diff
@@ -35,15 +40,25 @@ final class DoctrineDBALConnectionSourceIterator implements SourceIteratorInterf
     private int $position = 0;
 
     /**
-     * @var Result
+     * @var Result|DriverResult
      */
     private $result;
 
     /**
-     * @param mixed[] $parameters
+     * @param Connection|DriverConnection $connection
+     * @param mixed[]                     $parameters
      */
-    public function __construct(Connection $connection, string $query, array $parameters = [])
+    public function __construct($connection, string $query, array $parameters = [])
     {
+        if (!$connection instanceof Connection) {
+            @trigger_error(sprintf(
+                'Passing an instance of %s as argument 1 is deprecated since sonata-project/exporter 2.13'
+                .' and will not work in 3.0. You MUST pass an instance of %s instead',
+                \get_class($connection),
+                Connection::class
+            ), \E_USER_DEPRECATED);
+        }
```

Comment:
```suggestion
        if ($connection instanceof DriverConnection) {
            @trigger_error(sprintf(
                'Passing an instance of %s as argument 1 is deprecated since sonata-project/exporter 2.13'
                .' and will not work in 3.0. You MUST pass an instance of %s instead',
                \get_class($connection),
                Connection::class
            ), \E_USER_DEPRECATED);
        } elseif (!$connection instanceof Connection) {
            throw new \TypeError(sprintf(
                '%s: Argument 1 is expected to be an instance of %s, got %s.',
                __METHOD__,
                Connection::class,
                get_class($connection)
            ));
        }
```

You can still throw if `$connection` is not `Connection|DriverConnection` as you did before.

Reply:
Indeed. I just need to do differently since in DBAL 2 the connection implements DriverConnection
https://github.com/doctrine/dbal/blob/v2.10.1/lib/Doctrine/DBAL/Connection.php#L37